build_tools/cmake/iree_copts.cmake

```diff
@@ -456,7 +456,7 @@ endif()
 set(MLIR_TABLEGEN_EXE mlir-tblgen)
 # iree-tblgen is not defined using the add_tablegen mechanism as other TableGen
 # tools in LLVM.
-iree_get_executable_path(IREE_TABLEGEN_EXE iree-tblgen)
+iree_get_executable_path(IREE_TABLEGEN_EXE iree::tools::iree-tblgen)
```

Comment:
this error from presubmit is legit:

```
CMake Error at build_tools/cmake/iree_macros.cmake:129 (message):
  Could not find 'iree::tools::iree-tblgen' at
  '/tmpfs/src/github/iree/build-android/../build-host/install/bin/iree::tools::iree-tblgen'.
  Ensure that IREE_HOST_BINARY_ROOT points to installed binaries.
Call Stack (most recent call first):
  build_tools/cmake/iree_copts.cmake:459 (iree_get_executable_path)
  CMakeLists.txt:216 (include)
```

`iree_get_executable_path` looks for either a target or an installed executable with the given name:

https://github.com/google/iree/blob/66e44e9662d0e53f15ee50272aabbf27f533af3b/build_tools/cmake/iree_macros.cmake#L114-L133

Some ideas:

* pass the target _and_ the name, and let the function do the switch
* strip the package (`iree::tools::`) in the function and check for an executable, assuming the target name matches the executable name (as we do now)

Reply:
+1 to global aliases for binaries

Reply:
Done (restored binary global aliases). We might at some point want to make this opt-in for binaries so we just get tools and not "integration_test" (https://github.com/google/iree/blob/cc4c05d7e9a64607c526b75eb14acc619b47b36c/experimental/bindings/java/com/google/iree/tests/CMakeLists.txt#L30), but this is the behavior we already had, so I'm going to punt on that for now